kafka-rest/src/main/java/io/confluent/kafkarest/controllers/PartitionManagerImpl.java

```diff
@@ -73,7 +79,23 @@ public CompletableFuture<Optional<Partition>> getPartition(
       String clusterId, String topicName, int partitionId) {
     return topicManager
         .getTopic(clusterId, topicName)
-        .thenApply(topic -> checkEntityExists(topic, "Topic %s cannot be found.", topic))
+        .handle(
+            (value, exception) -> {
+              if (exception != null) {
+                log.error("Caught exception while get topic from cluster", exception);
```

Comment:
Does this log line duplicate the new log line in KafkaFutures ? I don't think we need?

It would be useful to see the full log output after this PR change - can you include it in the main thread of the PR please.

Reply:
The exception output is 
```
[2022-09-08 22:11:25,656] ERROR Caught Exception on completion of KafkaFuture (io.confluent.kafkarest.common.KafkaFutures:47)
java.util.concurrent.CompletionException: org.apache.kafka.common.errors.UnknownTopicOrPartitionException: This server does not host this topic-partition.
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at org.apache.kafka.common.internals.KafkaCompletableFuture.kafkaCompleteExceptionally(KafkaCompletableFuture.java:49)
	at org.apache.kafka.common.internals.KafkaFutureImpl.completeExceptionally(KafkaFutureImpl.java:130)
	at org.apache.kafka.common.KafkaFuture.lambda$allOf$2(KafkaFuture.java:93)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at org.apache.kafka.common.internals.KafkaCompletableFuture.kafkaCompleteExceptionally(KafkaCompletableFuture.java:49)
	at org.apache.kafka.common.internals.KafkaFutureImpl.completeExceptionally(KafkaFutureImpl.java:130)
	at org.apache.kafka.clients.admin.KafkaAdminClient$5.handleResponse(KafkaAdminClient.java:1980)
	at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.handleResponses(KafkaAdminClient.java:1268)
	at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.processRequests(KafkaAdminClient.java:1421)
	at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.run(KafkaAdminClient.java:1344)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.kafka.common.errors.UnknownTopicOrPartitionException: This server does not host this topic-partition.
```

I also added it in the main thread.